### ISSUE TITLE
Option to predict tagger model without SpacyWordSplitter?

### ISSUE THREAD

**Issue by sebastianGehrmann** [allenai/allennlp#917]
I am training a SimpleTagger instance and when trying to use the trained model to predict unseen data, I keep running into the problem that the output has more words than the gold. 
I looked into the code and instances of the class `sentence_tagger` always use the SpacyTokenizer (see here: https://github.com/allenai/allennlp/blob/master/allennlp/service/predictors/sentence_tagger.py#L22). 

Since I am formatting the training data so that I have a `word###tag` format, I am assuming that the splitter is not applied here. Assuming that the test tgt is formatted in a spacy-compliant way seems like a pretty big assumption to me. 
Since you already have the JustSpaceTokenizer here (https://github.com/allenai/allennlp/blob/master/allennlp/data/tokenizers/word_splitter.py#L128), I am wondering if there a method exists to tell the predictor to use that one (without monkeypatching the class on my end). 
If not, I suggest adding a tokenizer option to the config file.

**Comment by matt-gardner** [comment]
What are you trying to do?  Are you trying to evaluate a model's performance on data that you have labels for?  Or make predictions on data for which you don't have labels?  It sounds like you're trying to evaluate performance ("output has more words than the gold"), but you're using our infrastructure for making predictions on data that you don't have labels for (the `Predictor`).  The `Predictor` exists largely for use with a demo, where you wouldn't have any labels, which is why it's in the `service` module.

If you just want to make predictions and compute accuracy on unseen data that's formatted the same as your training data, use the `evaluate` command.

**Comment by sebastianGehrmann** [comment]
The tl;dr of my problem is that I want to predict on already tokenized, unseen data. 

Maybe I am not quite understanding what the evaluation command does. Following the documentation, my understanding is that `evaluate` needs an input formatted the same way as the training data (including targets) and outputs accuracy metrics. I have not seen a way of using it as a substitute for `predict` such that I input a text without targets and get the json output file. 

So far, I have played with the `predict` mode on my validation set to analyze the model output. That's why I have access to gold labels and made the comment about number of predictions. Eventually I would like to predict using a dataset without having the labels. However, I need to keep the tokenization the same way as in training - my model is trained using a different tokenization and using the spacy tokenizer on top of my test set leads to worse results. 

Right now I just modified the source files above to not use the tokenizer and reinstalled the package.
